src/grammar/dml.rs

```diff
@@ -0,0 +1,57 @@
+use super::parse_where_clause;
+use crate::parser::Parser;
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    parse_delete(p);
+}
+
+pub(crate) fn parse_delete(p: &mut Parser) {
```

Comment:
Please add this to `parse_stmt`, which in turn is used by `parse_block`, including tests.
`parse_block` is used by all kinds of constructs, such as procedures and packages.